lib/Doctrine/ORM/Query.php

```diff
@@ -403,10 +406,17 @@ private function processParameterMappings($paramMappings)
                 $value = array_keys(HierarchyDiscriminatorResolver::resolveDiscriminatorsForClass($value, $this->em));
             }
 
+            if (is_object($value) && $this->em->getMetadataFactory()->hasMetadataFor(get_class($value))) {
```

Comment:
Can the block deciding `$type` and `$value` be isolated into a `private` method?

Reply:
`$value` may also be a proxy, then this code will not work

Reply:
> 
> 
> Can the block deciding `$type` and `$value` be isolated into a `private` method?

I'll work on this later, for now "it works"... :-)